src/mpid/ch4/netmod/ofi/ofi_types.h

```diff
@@ -73,18 +73,18 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_OFI_idata_get_error_bits(uint64_t idata)
 }
 
 
-#define MPIDI_OFI_PROTOCOL_BITS (3)     /* This is set to 3 even though we actually use 4. The ssend
-                                         * ack bit needs to live outside the protocol bit space to avoid
-                                         * accidentally matching unintended messages. Because of this,
-                                         * we shift the PROTOCOL_MASK one extra bit to the left to take
-                                         * the place of the empty SSEND_ACK bit. */
+#define MPIDI_OFI_PROTOCOL_BITS (2)     /* This is set to 2 even though we actually use 4. The ssend
+                                         * ack bit and dynproc bit need to live outside the protocol
+                                         * bit space to avoid accidentally matching unintended
+                                         * messages. Because of this, we shift the PROTOCOL_MASK
+                                         * two extra bits to the left. */
 
 #if MPIDI_OFI_ENABLE_RUNTIME_CHECKS == MPIDI_OFI_ON
 #define MPIDI_OFI_SYNC_SEND_ACK      (1ULL << (MPIDI_OFI_CONTEXT_BITS + MPIDI_OFI_SOURCE_BITS + MPIDI_OFI_TAG_BITS))
-#define MPIDI_OFI_SYNC_SEND          (2ULL << (MPIDI_OFI_CONTEXT_BITS + MPIDI_OFI_SOURCE_BITS + MPIDI_OFI_TAG_BITS))
-#define MPIDI_OFI_DYNPROC_SEND       (4ULL << (MPIDI_OFI_CONTEXT_BITS + MPIDI_OFI_SOURCE_BITS + MPIDI_OFI_TAG_BITS))
+#define MPIDI_OFI_DYNPROC_SEND       (2ULL << (MPIDI_OFI_CONTEXT_BITS + MPIDI_OFI_SOURCE_BITS + MPIDI_OFI_TAG_BITS))
+#define MPIDI_OFI_SYNC_SEND          (4ULL << (MPIDI_OFI_CONTEXT_BITS + MPIDI_OFI_SOURCE_BITS + MPIDI_OFI_TAG_BITS))
```

Comment:
Is this swapping intentional?

Reply:
Yes, I wanted to keep the bits used for matching together.